projects/plugins/jetpack/_inc/lib/core-api/wpcom-endpoints/class-wpcom-rest-api-v2-endpoint-block-editor-assets.php

```diff
@@ -474,18 +500,363 @@ private function unregister_disallowed_plugin_assets() {
 	 * @return bool True if the asset is a core or Gutenberg asset, false otherwise.
 	 */
 	private function is_core_or_gutenberg_asset( $src ) {
+		return $this->is_core_asset( $src ) || $this->is_gutenberg_asset( $src );
+	}
+
+	/**
+	 * Check if an asset is a core WordPress asset.
+	 *
+	 * @param string $src The asset source URL.
+	 * @return bool True if the asset is a core WordPress asset, false otherwise.
+	 */
+	private function is_core_asset( $src ) {
 		if ( ! is_string( $src ) ) {
 			return false;
 		}
 
 		return empty( $src ) ||
-			$src[0] === '/' ||
-			str_contains( $src, 'wp-includes/' ) ||
-			str_contains( $src, 'wp-admin/' ) ||
-			str_contains( $src, 'plugins/gutenberg/' ) ||
+			str_contains( $src, '/wp-includes/' ) ||
+			str_contains( $src, '/wp-admin/' );
+	}
+
+	/**
+	 * Check if an asset is a Gutenberg plugin asset.
+	 *
+	 * @param string $src The asset source URL.
+	 * @return bool True if the asset is a Gutenberg plugin asset, false otherwise.
+	 */
+	private function is_gutenberg_asset( $src ) {
+		if ( ! is_string( $src ) ) {
+			return false;
+		}
+
+		return str_contains( $src, 'plugins/gutenberg/' ) ||
```

Comment:
In some setups sites might be placing `plugins` folder but a different url. You could use the `plugins_url()` function to find the more reliable url. which is also that Gutenberg uses .